src/tests/conformance/invalid_message/disconnect.rs

```diff
@@ -0,0 +1,303 @@
+//! Contains test cases which cover ZG-CONFORMANCE-011.
+//!
+//! The node disconnects for trivial (non-fuzz, non-malicious) cases.
+//!
+//! [`Ping`](Message::Ping) timeout.
+//! [`Pong`](Message::Pong) with wrong [`Nonce`].
+//! [`GetData`](Message::GetData) containing both [`Block`][inv_block] and [`Tx`][inv_tx] types in [`InvHash`][inv_hash].
+//! [`Inv`](Message::Inv) containing both [`Block`][inv_block] and [`Tx`][inv_tx] types in [`InvHash`][inv_hash].
+//! [`Addr`](Message::Addr) containing a [`NetworkAddr`][net_addr] *without* a  timestamp.
+//!
+//! [inv_block]: crate::protocol::payload::inv::ObjectKind::Block
+//! [inv_tx]: crate::protocol::payload::inv::ObjectKind::Tx
+//! [inv_hash]: crate::protocol::payload::inv::InvHash
+//! [net_addr]: crate::protocol::payload::addr::NetworkAddr
+
+use std::{
+    io,
+    net::{IpAddr, Ipv4Addr, SocketAddr},
+    time::Duration,
+};
+
+use crate::{
+    protocol::{
+        message::{
+            constants::{ADDR_COMMAND, HEADER_LEN},
+            Message, MessageHeader,
+        },
+        payload::{addr::NetworkAddr, block::Block, codec::Codec, Inv, Nonce},
+    },
+    setup::node::{Action, Node},
+    tools::{
+        message_filter::{Filter, MessageFilter},
+        synthetic_node::{PingPongError, SyntheticNode},
+    },
+};
+
+const DC_TIMEOUT: Duration = Duration::from_secs(1);
+
+#[tokio::test]
+#[ignore = "ZCashd ping timeout is 20min"]
```

Comment:
I don't know if this is good - since it will now clash with our internal protocol tests (I think?).

I do want to include the test but I don't quite know how else.

Reply:
`--lib` wil run our entire test suite (since the `tests` are in `lib`). Feature flags might be an option. Another possibility is to have two crates: `core` and `suite`, though I'm hesitant to make structural changes of that magnitude this late in the grant. This would best be left for later. 

Simply removing `#[tokio::test]` and adding a comment is probably best, especially since no one is ever actually going to run this test case. For that reason I'm actually partial to removing the case entirely. (As long as the reason for doing so is documented somewhere obvious.)

Reply:
Okay, will add a comment and remove it.